Select2 v4 support

grappler opened the pull request:
Adds Select2 v4 support.

I have tested post select options and simple select option. Multiple and single select items work too.

Resolves #363

*I have:*
 - [x] Run WordPress VIP PHPCS check and code parses without errors
 - [ ] Added any new PHPUnit tests
 - [ ] Run PHPUnit and all tests are passing after adding any necessary new tests
 - [x] Tested feature/bugfix on ~~single and~~ multisite install

@mikeselander

grappler committed:
Add support for select2 v4

mikeselander commented:
Hey Ulrich just so you know I've seen this but thing picked up so it'l be a little bit before I can review in depth. Thanks again for doing all this!

ocean90 committed:
Center the file upload button vertically

ocean90 committed:
Update minified CSS file

ocean90 committed:
Avoid jumping buttons on :active.

WordPress has a transform rule which moves the button by 1px on :active.

grappler committed:
Add support for select2 v4

Update after rebase. classes.fields.php was deleted in favour for single files.

mikeselander commented:
As discussed at WCEU, we unfortunately won't be adding enhancements like this to HMCMB.
